libevmasm/PeepholeOptimiser.cpp

```diff
@@ -428,6 +428,36 @@ struct TruthyAnd: SimplePeepholeOptimizerMethod<TruthyAnd>
 	}
 };
 
+// if(a == true) -> if(a)
```

Comment:
Will this work the other way around, i.e. `true == a`?

Reply:
Oh good spot, this indeed won't work for `true == a`, also need to add similar rules for `false` checks. Thanks for the other suggestions as well. Will get these done, pushed the draft to make sure I'm headed in the right direction :)

Reply:
No worries. In any case, @ekpyron and @chriseth are experts in this domain, so they'll likely provide a more detailed review. Be patient though, since they're quite busy. Just make sure to rebase against fresh develop from time to time to keep the PR up to date.